### PR TITLE
add crosspath, BinaryTree, and RoachGraph

### PR DESCRIPTION
**Pull request by jpfairbanks**
graphs from Guattery Miller 1998 on the quality of spectral seperators.
includes tests.

### CODE REVIEW

File: src/smallgraphs.jl [hunk 1]
```diff
@@ -524,3 +524,38 @@ function TutteGraph()
     ]
     return _make_simple_undirected_graph(46,e)
 end
+
+"""create a binary tree with k-levels vertices are numbered 1:2^levels-1"""
+function BinaryTree(levels)
+    g = Graph(2^levels-1)
+    for i in 0:levels-2
+        for j in 2^i:2^(i+1)-1
+            j; add_edge!(g, j, 2j)
```
Comment (sbromberger) [inline src/smallgraphs.jl]: what's that `j;` hanging out there for?

File: src/smallgraphs.jl [hunk 2]
```diff
@@ -524,3 +524,38 @@ function TutteGraph()
...
+            add_edge!(g, j, 2j+1)
+        end
+    end
+    return g
+end
+
+"""create a double complete binary tree with k-levels 
+used as an example for spectral clustering by Guattery and Miller 1998."""
+function doubletree(levels)
```
Comment (sbromberger) [inline src/smallgraphs.jl]: bikeshed the name: `DoubleBinaryTree(levels::Int)` please?

File: src/smallgraphs.jl [hunk 3]
```diff
@@ -524,3 +524,38 @@ function TutteGraph()
     ]
     return _make_simple_undirected_graph(46,e)
 end
+
+"""create a binary tree with k-levels vertices are numbered 1:2^levels-1"""
+function BinaryTree(levels)
```
Comment (sbromberger) [inline src/smallgraphs.jl]: we need to typecast `levels::Int` here.

File: src/operators.jl [hunk 4]
```diff
@@ -151,6 +151,33 @@ function join(g::Graph, h::Graph)
...
+#=     for i in 1:len =#
+#=         k = nv(g) =#
+#=         g = blkdiag(g,g) =#
+#=         for v in 1:k =#
+#=             add_edge!(g, v, v+k) =#
+#=         end =#
+#=     end =#
+#=     return g =#
+#= end =#
+
```
Comment (sbromberger) [inline src/operators.jl]: do we need this comment block?

File: src/LightGraphs.jl [hunk 5]
```diff
@@ -84,6 +85,7 @@ KrackhardtKiteGraph, MoebiusKantorGraph, OctahedralGraph,
 PappusGraph, PetersenGraph, SedgewickMazeGraph,
 TetrahedralGraph, TruncatedCubeGraph,
 TruncatedTetrahedronGraph, TruncatedTetrahedronDiGraph, TutteGraph,
+BinaryTree, RoachGraph,
```
Comment (sbromberger) [inline src/LightGraphs.jl]: where's double binary tree?

File: test/smallgraphs.jl [hunk 6]
```diff
@@ -94,3 +94,39 @@ g = TruncatedTetrahedronDiGraph()
...
+#=  [10,7,6] =# 
+#=  [11,10,7] =#
+#=  [8,9,12] =# 
+#=  [9,12] =#   
+#=  [10,11] =#  
+I = [ 1,2,3,3,4,4,5,5,6,6,7,7,7,8,8,8,9,9,9,10,10,10,11,11,12,12 ]
+J = [ 3,4,1,5,2,6,3,7,4,8,9,8,5,10,7,6,11,10,7,8,9,12,9,12,10,11 ]
+V = ones(Int, length(I))
+Adj = sparse(I,J,V)
+@test Adj == sparse(rg3)
```
Comment (sbromberger) [inline test/smallgraphs.jl]: test for double binary trees?

**Comment by sbromberger** [comment]
Looks very good. Can you resolve the conflicts?

File: src/smallgraphs.jl [hunk 7]
```diff
@@ -524,3 +524,38 @@ function TutteGraph()
     ]
     return _make_simple_undirected_graph(46,e)
 end
+
+"""create a binary tree with k-levels vertices are numbered 1:2^levels-1"""
+function BinaryTree(levels::Int)
```
Comment (sbromberger) [inline src/smallgraphs.jl]: If you could rebase this off of the datasets branch, that'd be great.

**Comment by jpfairbanks** [comment]
I just merged master into this branch, the conflict was just in the export list.

**Comment by sbromberger** [comment]
I will move these into Datasets at some other point.

**Pull request merged**